subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -171,14 +171,8 @@ int bt_csip_sef(const uint8_t k[BT_CSIP_CRYPTO_KEY_SIZE], const uint8_t sirk[BT_
 
 	LOG_DBG("SIRK %s", bt_hex(sirk, BT_CSIP_SIRK_SIZE));
 
-	if (IS_ENABLED(CONFIG_LITTLE_ENDIAN)) {
-		/* Swap because aes_cmac is big endian
-		 * and we are little endian
-		 */
-		sys_memcpy_swap(k1_tmp, k, sizeof(k1_tmp));
-	} else {
-		(void)memcpy(k1_tmp, k, sizeof(k1_tmp));
-	}
+	/* Swap because aes_cmac is big endian and Bluetooth is little endian */
```

Comment:
I think it's a bit misleading to call Bluetooth little endian. 

Not completely sure to properly define/describe this, but
Zephyr stores the LTK as little-endian, and `bt_csip_sef` takes the LTK as `k` for LE (but the Link key for bredr). 

So the endianess of `k` doesn't really depend on how Bluetooth transmits this on air, but how Zephyr stores it. 

`k1` is the one that calls `bt_crypto_aes_cmac`, which for reasons I don't know, expect the `in` parameter to be big-endian (but not the key?), and the LTK is the `in` parameter in this case. 

I don't recall why the swapping is done here in `bt_csip_sef` instead of in `k1`, but it might fit better in the `k1` than here.

Reply:
As far as I understand for `k1` the key given to `bt_crypto_aes_cmac` is the `salt` given by `s1` which also uses `bt_crypto_aes_cmac` and thus is already big-endian.

Reply:
Correct

Reply:
This kind of confusion is why I would usually prefer to label the different endianness buffers by the `_be` suffix.

```
	const uint8_t m_be[] = {'S', 'I', 'R', 'K', 'e', 'n', 'c'};
	const uint8_t p_be[] = {'c', 's', 'i', 's'};
	uint8_t s1_be_out[BT_CSIP_CRYPTO_SALT_SIZE];
	uint8_t k1_be_out[BT_CSIP_CRYPTO_KEY_SIZE];
	uint8_t k1_be[BT_CSIP_CRYPTO_KEY_SIZE];
```
This would be better names IMO.

Example: 34e66db776bc4e19aefa9c51ad7657427d54f779

Reply:
Although it's a bit of a misnomer to use the term "endianess" on byte arrays, I'm in favor of using @joerchan suggestion to label the variables with their byte order. In the controller we also have an internal HAL API towards ECB encryption that uses the `le` and `be` labeling (see subsys/bluetooth/controller/hal/ecb.h):

```
void ecb_encrypt_be(uint8_t const *const key_be, uint8_t const *const clear_text_be,
                    uint8_t * const cipher_text_be);
void ecb_encrypt(uint8_t const *const key_le, uint8_t const *const clear_text_le,
                 uint8_t * const cipher_text_le, uint8_t * const cipher_text_be);
```

There raw encryption function even has the possibility to provide the cipher result in the byte order you need.

Reply:
> Although it's a bit of a misnomer to use the term "endianess" on byte arrays, 

Except when the byte arrays are reflecting values larger than 64-bits or strings, which is the case here :D